src/service/resources-impl.js

```diff
@@ -435,6 +435,69 @@ export class Resources {
   }
 
   /**
+   * Runs the specified subtree "show" mutation and ensures that measures
+   * and layouts performed for the affected elements.
+   * @param {!Element} root
+   * @param {function()} mutator
+   * @return {!Promise}
+   */
+  showSubtree(root, mutator) {
```

Comment:
@cramforce on this advise, I reimplemented this functionality. I think it looks a lot better. The main changes are:

1. Resources pass can now identifying elements that need to be unloaded when they become invisible. This provides a much better coverage since it'd catch unloads from all sources this way.
2. I still use `relayoutTop` and `requestMeasure` methods. It's true that we could simply do "relayoutAll" instead. However, I dug through my notes and there's definitely a non-trivial cost for re-measurements despite the fact that they are in the same measurement phase. This cost is pretty low in Chrome and Safari (but not that small) and it's very high in Mozilla. I can provide more details if you'd like to see. The bottom line is - if we can reduce number of remeasures, we should do that.
3. Now Resources has a single `mutateElement` method. It doesn't care what the operation really is but can do the right thing in most typical cases.

I will also send an independent PR that would show case `hide` action using this API. But I think it should already be clear from this PR.

PTAL.